app/services/idv/steps/inherited_proofing/verify_wait_step_show.rb

```diff
@@ -13,6 +14,18 @@ def self.analytics_optional_step_event
         end
 
         def call
+          # NOTE: The only way throttle.attempts will be > 0, is if our job to
+          # retrieve the user's PII failed on the first attempt (from the previous
+          # :agreement step), and the user subsequently clicks the "Retry" button on the
+          # Inherited Proofing (IP) Warning UI. The IP Warning UI "Retry" button
+          # routes to this step, allowing the user to retry PII retrieval from the Service
+          # Provider IdentityConfig.store.inherited_proofing_max_attempts times, before
+          # the IP Error UI is displayed, allowing no further retry attempts. #enqueue_job
+          # checks to make sure that a job is not already in progress before enqueuing a
+          # new job to retrieve the users PII, so there is no danger here of jobs being
+          # added to the queue while polling the async state is in progress.
+          enqueue_job if throttle.attempts > 0
```

Comment:
Additionally, does this mean that if a user does not click "retry" the client will continue polling to process async state even though we already know the request failed?

Reply:
Also, I think that if it polls again or the user refreshes the job will be automatically re-enqueued. If that is not the case then it is not obvious to me how we avoid that.

Reply:
Yes, will have to look in to using POST instead; good catch.

>... does this mean that if a user does not click "retry" the client will continue polling to process async state even though we already know the request failed?

No, because the "Retry" view is redirected _to_ by the `VerifyWaitStepShow` step. When the user clicks the "Retry" button from the "Retry" view, it redirects _back_ to the `VerifyWaitStepShow` step view, re-enqueues a new job, and starts to poll the job state. 

> Also, I think that if it polls again or the user refreshes the job will be automatically re-enqueued. If that is not the case then it is not obvious to me how we avoid that.

`UserPiiJobInitiator#enqueue_job` does not enqueue a new job if it already exists.

Reply:
Addressed.